squirtgun-platform-fabric/src/main/java/me/lucyy/squirtgun/fabric/FabricPlatform.java

```diff
@@ -0,0 +1,214 @@
+ * the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+ * LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+ * WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package me.lucyy.squirtgun.fabric;
+
+import com.google.common.collect.Lists;
+import me.lucyy.squirtgun.fabric.task.FabricTaskScheduler;
+import me.lucyy.squirtgun.platform.AuthMode;
+import me.lucyy.squirtgun.platform.EventListener;
+import me.lucyy.squirtgun.platform.Platform;
+import me.lucyy.squirtgun.platform.audience.SquirtgunUser;
+import me.lucyy.squirtgun.plugin.SquirtgunPlugin;
+import net.fabricmc.fabric.api.event.lifecycle.v1.ServerLifecycleEvents;
+import net.fabricmc.loader.api.FabricLoader;
+import net.kyori.adventure.platform.fabric.FabricServerAudiences;
+import net.kyori.adventure.text.Component;
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.server.command.ServerCommandSource;
+import net.minecraft.server.dedicated.MinecraftDedicatedServer;
+import net.minecraft.server.network.ServerPlayerEntity;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Objects;
+import java.util.UUID;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.logging.Logger;
+
+/**
+ * Squirtgun Platform implementation for the Fabric mod loader.
+ */
+public class FabricPlatform implements Platform {
+
+	private static final Logger LOGGER = Logger.getLogger(FabricPlatform.class.getSimpleName());
+	private static final List<String> PROXY_BRIDGING_MODS = List.of("fabricproxy", "fabricproxy-lite");
+
+	private MinecraftDedicatedServer server;
+	private FabricServerAudiences audiences;
+	private List<FabricPlayer> onlinePlayers = List.of();  // default to empty list until server has started
+	private final FabricTaskScheduler taskScheduler;
+	private final FabricConsoleWrapper consoleWrapper;
+	private final FabricListenerAdapter listenerAdapter = new FabricListenerAdapter();
+
+	/**
+	 * Create the Squirtgun's Fabric "platform" for the server.
+	 *
+	 * <p>The server instance passed <b>must</b> be a {@link MinecraftDedicatedServer}.</p>
+	 *
+	 * @param server server instance to work with
+	 */
+	public FabricPlatform(final @NotNull MinecraftServer server) {
+		this.server = validateState(MinecraftDedicatedServer.class::isInstance, server, MinecraftDedicatedServer.class::cast, "Squirtgun cannot be used on the client!");
```

Comment:
why this restriction?

Reply:
In some part it is not strictly necessary, but for something like getting the auth mode
https://github.com/lucyy-mc/Squirtgun/blob/7d9e580ab9d77a728aaa35a1e7f56360b5ce1239/squirtgun-platform-fabric/src/main/java/me/lucyy/squirtgun/fabric/FabricPlatform.java#L131
The "server properties" only exists on the [dedicated server](https://maven.fabricmc.net/docs/yarn-1.17+build.13/net/minecraft/server/dedicated/DedicatedServer.html#getProperties()), not the integrated one. The integrated server technically is always online since there is no `online-mode` option anywhere, let alone one connected to a proxy server. Can put a simple `instanceof` for that I guess? Not that it's an accurate measure.

Reply:
ah alright thats fine, thanks for clarifying